client/ftp-webclient/src/main/kotlin/ebi/ac/uk/ftp/FTPClientPool.kt

```diff
@@ -89,11 +94,24 @@ internal class FTPClientPool(
             ftpUrl: String,
             ftpPort: Int,
             ftpRootPath: String,
+            defaultTimeout: Long,
+            connectionTimeout: Long,
         ): GenericObjectPool<FTPClient> {
-            val factory = FTPClientFactory(ftpUser, ftpPassword, ftpUrl, ftpPort, ftpRootPath)
+            val factory =
+                FTPClientFactory(
+                    ftpUser,
+                    ftpPassword,
+                    ftpUrl,
+                    ftpPort,
+                    ftpRootPath,
+                    defaultTimeout,
+                    connectionTimeout,
+                )
             var connections = GenericObjectPool(factory)
             connections.minIdle = MIN_CONNECTION
             connections.testOnBorrow = true
+            connections.testOnReturn = true
```

Comment:
what is the porpose of this two? testOnReturn and  blockWhenExhausted

Reply:
`testOnReturn` checks the object is valid before returning it to the pool
`blockWhenExhausted` isn't required here. I realized about that later and removed it in my latest commit